server-spi-private/src/main/java/org/keycloak/services/scheduled/ScheduledTaskRunner.java

```diff
@@ -46,32 +46,23 @@ public ScheduledTaskRunner(KeycloakSessionFactory sessionFactory, ScheduledTask
 
     @Override
     public void run() {
-        KeycloakSession session = sessionFactory.create();
-        try {
+        try (KeycloakSession session = sessionFactory.create()) {
+            session.getTransactionManager().begin();
             if (transactionLimit != 0) {
                 KeycloakModelUtils.setTransactionLimit(sessionFactory, transactionLimit);
             }
             runTask(session);
         } catch (Throwable t) {
             logger.errorf(t, "Failed to run scheduled task %s", task.getClass().getSimpleName());
-
-            session.getTransactionManager().rollback();
```

Comment:
If there is an error how is the transaction now rolled-back?

Reply:
This is done in `close` method invoked at the end of the try-with-resources block, specifically in `DefaultKeycloakSession.closeTransactionManager`

Reply:
That looks like it's committing the TX, not rolling it back?